src/main/java/org/webshar/hrms/service/EmployeeService.java

```diff
@@ -65,10 +65,19 @@ public Employee updateEmployee(final Long id, EmployeeUpdateRequest employeeUpda
       throws EntityNotFoundException, BadRequestException {
     Employee employeeToUpdate = employeeRepository.findById(id)
         .orElseThrow(() -> new EntityNotFoundException(ErrorMessageConstants.EMPLOYEE_BY_ID_NOT_FOUND));
-    Employee updatedEmployee = employeeBuilder.buildFromRequest(employeeToUpdate, employeeUpdateRequest);
+    Optional<Employee> reportTo = getReportsToEmployee(employeeUpdateRequest.getReportsTo());
+    Employee updatedEmployee = employeeBuilder.buildFromRequest(employeeToUpdate, employeeUpdateRequest, reportTo);
     return employeeRepository.save(updatedEmployee);
   }
 
+  private Optional<Employee> getReportsToEmployee(Long reportsTo) {
+    Optional<Employee> reportTo = null;
+    if (reportsTo != null) {
+      reportTo = employeeRepository.findById(reportsTo);
+    }
+    return reportTo;
```

Comment:
This is not yet done. Need to correct first statement to `Optional<Employee> reportTo = Optional.empty();`
See sonar lint warning for more details.

Reply:
Yeah missed that in last comment done